sycl/source/detail/kernel_bundle_impl.hpp

```diff
@@ -68,6 +140,27 @@ class kernel_bundle_impl {
     MDeviceImages.erase(It, MDeviceImages.end());
   }
 
+  // C'tor matches sycl::join API
+  kernel_bundle_impl(const std::vector<detail::KernelBundleImplPtr> &Bundles) {
+    MContext = Bundles[0]->MContext;
+    for (const detail::KernelBundleImplPtr &Bundle : Bundles) {
```

Comment:
Probably I didn't get the idea. But if I understand it correctly it falls to the same problem we discussed previously - that using std::set here would require repacking values back to vector since there are required APIs which return values in std::vector.